contracts/message-bus/MessageInbox.sol

```diff
@@ -0,0 +1,216 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./MessageBox.sol";
+import "./StateRootI.sol";
+import "./Proof.sol";
+
+contract MessageInbox is MessageBox, Proof {
+
+    /** Mapping to indicate that message hash exists in inbox. */
+    mapping(bytes32 => bool) public inbox;
```

Comment:
Just wondering if we can make this backward compaitable and make it work with existing gateways? 

Maybe Then `Bool` is not enough. Do we need MessageStatus enum?